app/src/main/java/com/example/concordia_campus_guide/Fragments/LocationFragment/LocationFragment.java

```diff
@@ -330,11 +332,17 @@ public boolean setupBuildingMarkerClickListener(GoogleMap map) {
         map.setOnMarkerClickListener(new GoogleMap.OnMarkerClickListener() {
             @Override
             public boolean onMarkerClick(Marker marker) {
-                Building building = mViewModel.getBuildingFromeCode(marker.getTag().toString());
-                //TODO: Make function that pops up the info card for the building (via the building-code)
-                String buildingCode = (marker.getTag()).toString();
-                ((MainActivity) getActivity()).showInfoCard(buildingCode);
-                onBuildingClick(building);
+                if (marker.getTag() != null) {
+                    if (marker.getTag().toString().contains(POI_TAG)) {
+                        //TODO: Use the directions api to get directions to this point of interest
```

Comment:
why do we need to do this? Can't we use the same logic that is already used in the routes activity? Since a POI is a place, we should be able to use the SelectToFromState

Reply:
Yes we can!

Reply:
Nevermind, I got it!